### PR TITLE
doc: releases: introduce release notes and migration guide docs for 4.1

### PR DESCRIPTION
**Pull request by kartben**
https://builds.zephyrproject.io/zephyr/pr/80859/docs/releases/index.html
https://builds.zephyrproject.io/zephyr/pr/80859/docs/releases/release-notes-4.1.html
https://builds.zephyrproject.io/zephyr/pr/80859/docs/releases/migration-guide-4.1.html

This introduces the release notes and migration guide for 4.1.0 earlier than we typically do, so that people have a placeholder to start adding content as they line up pull requests for 4.1. 
The two documents are currently orphan and not visible from the main documentation as this would confuse users of 4.0.

See commit messages for other minor touch-up

### CODE REVIEW

**Review by Thalley — APPROVED** [review]
Thanks :)

File: doc/releases/index.rst
Comment (tomi-font) [inline doc/releases/index.rst:120]:
```suggestion
End-of-life releases
```

File: doc/releases/eol_releases.rst
Comment (tomi-font) [inline doc/releases/eol_releases.rst:26]:
```suggestion
Migration guides
```

File: doc/releases/eol_releases.rst
Comment (tomi-font) [inline doc/releases/eol_releases.rst:11]:
```suggestion
Release notes
```

File: doc/releases/index.rst
Comment (tomi-font) [inline doc/releases/index.rst:127]: Uhm, it seems that this effectively does nothing as no table of contents shows up: https://builds.zephyrproject.io/zephyr/pr/80859/docs/releases/index.html#end-of-life-releases

File: doc/releases/migration-guide-4.1.rst
Comment (tomi-font) [inline doc/releases/migration-guide-4.1.rst:5]:
```suggestion
Migration guide to Zephyr v4.1.0 (Working draft)
```

File: doc/releases/release-notes-4.1.rst
Comment (tomi-font) [inline doc/releases/release-notes-4.1.rst:5]:
```suggestion
Zephyr 4.1.0 (Working draft)
```

**Review by tomi-font — COMMENTED** [review]
Thanks for putting the links to the generation documentation in your PR description!

**Review by aescolar — APPROVED** [review]

File: doc/releases/index.rst
Comment (kartben) [reply doc/releases/index.rst:120]: The document already has consistent capitalization in all the headings so I would suggest to leave as-is

File: doc/releases/eol_releases.rst
Comment (kartben) [reply doc/releases/eol_releases.rst:26]: ditto

File: doc/releases/eol_releases.rst
Comment (kartben) [reply doc/releases/eol_releases.rst:11]: ditto

Comment (tomi-font) [reply doc/releases/eol_releases.rst:11]: Then you might want to align the capitalization of `End-of-life releases` at the beginning of this page.

File: doc/releases/index.rst
Comment (kartben) [reply doc/releases/index.rst:127]: it's basically only there to "mount" the document in the overall document hierarchy

Comment (tomi-font) [reply doc/releases/index.rst:127]: Ahh, my bad. I see in the document hierarchy there are two `End-of-life releases` levels. Could changing `:maxdepth:` to 2 make it so that `Release Notes` and `Migration Guides` appear in the hierarchy? Or rather maybe just remove the first title, [End-of-life releases](https://builds.zephyrproject.io/zephyr/pr/80859/docs/releases/eol_releases.html#end-of-life-releases), so that the subtitles appear directly in the hierarchy? Don't know. :thinking:

**Review by tomi-font — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
If I was a user that actively works on pinned releases I would find it very confusing to bump into documents that refers to future releases. I'd imagine one would open the releases directory and just look for the latest document available and take that, no one has to figure what releases are they actually looking at. I understand this is convenient for us but IMO it's somewhat missing the point. Anyway, your call.

**Comment by tomi-font** [comment]
@fabiobaltieri Those documents won't show up in the documentation (they are orphan). Are you talking about the Git repo itself? Maybe a more explicit warning than `Working draft` could be added? Or then, yet another possibility would be to have these future release notes/migration guide placed somewhere else, but that could bring some issues of its own.

**Comment by fabiobaltieri** [comment]
yeah my concern is about having the files in the repo, how about calling the actual file draft? like `release-notes-4.1-draft.rst`? then the situation would be obvious just from listing the file and I think git would have no problems tracking the rename across PRs (need to do some tests for that though)

**Comment by tomi-font** [comment]
> and I think git would have no problems tracking the rename across PRs (did some tests just in case and yeah works file, edit a file, rename it on main, rebase on main the edit transfer over to the new file with no problems)

Yeah works fine with Git, but will still require a PR rebase I imagine? Or will it also work with GitHub PRs for when the file is renamed?

**Comment by Thalley** [comment]
> yeah my concern is about having the files in the repo, how about calling the actual file draft? like `release-notes-4.1-draft.rst`? then the situation would be obvious just from listing the file and I think git would have no problems tracking the rename across PRs (did some tests just in case and yeah works file, edit a file, rename it on main, rebase on main the edit transfer over to the new file with no problems)

Is the idea then to remove the `draft` of the name once we start tagging release candidates, or when should the `draft` part be removed?

**Comment by kartben** [comment]
> > and I think git would have no problems tracking the rename across PRs (did some tests just in case and yeah works file, edit a file, rename it on main, rebase on main the edit transfer over to the new file with no problems)
> 
> Yeah works fine with Git, but will still require a PR rebase I imagine? Or will it also work with GitHub PRs for when the file is renamed?

IIRC GitHub will indicate a conflict and won't want to merge *but* a rebase would "just work" without requiring human intervention for it will correctly detect the rename

**Comment by fabiobaltieri** [comment]
> IIRC GitHub will indicate a conflict and won't want to merge _but_ a rebase would "just work" without requiring human intervention for it will correctly detect the rename

I don't think it would trigger a conflict, if the file is renamed in main the new PR would just apply to the new name once the PR is rebase-merged.

> Is the idea then to remove the `draft` of the name once we start tagging release candidates, or when should the `draft` part be removed?

I'd do it at the same time as when draft is dropped from the title?

**Review by mmahadevan108 — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Pull request merged**
